src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -62,16 +62,42 @@ public void storeMessage(Message message) {
             .setFilter(new Query.FilterPredicate("recipient", FilterOperator.EQUAL, recipient))
             .addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
+    messages = message(user);
+    }
+
+    return messages;
+  }
+  public List<Message> getAllMessages(){
+    List<Message> messages = new ArrayList<>();
+   
+    Query query = new Query("Message")
+      .addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
 
+    for (Entity entity : results.asIterable()) {
+      try {
+        List<Message> userMessages = message(user); //get all messages from specific user
+        int uMSize = userMessages.size(); 
+        for (int i = 0; i < uMSize; i++) { //add each message from specific user to list of all messages
+          Message message = userMessages.get(i);
+          messages.add(userMessages);
+        }  
+      } catch (Exception e) {
+        System.err.println("Error getting messages.");
+        System.err.println(entity.toString());
+        e.printStackTrace();
+      }
+    }
+  }
+  public List<Message> message(String user) {
```

Comment:
It doesn't look like you need the `user` parameter.  It is never used. It is only overwritten by the `user` inside the `for` loop.